hudi-io/src/main/java/org/apache/hudi/storage/HoodieStorage.java

```diff
@@ -267,7 +270,7 @@ public final void createImmutableFileInPath(StoragePath path,
 
       if (content.isPresent() && needTempFile) {
         StoragePath parent = path.getParent();
-        tmpPath = new StoragePath(parent, path.getName() + TMP_PATH_POSTFIX);
+        tmpPath = new StoragePath(parent, path.getName() + "." + UUID.randomUUID());
         fsout = create(tmpPath, false);
```

Comment:
Here is the Hadoop filesystem atomicity guarantees: https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/filesystem/introduction.html#Core_Expectations_of_a_Hadoop_Compatible_FileSystem